### PR TITLE
added localized navmesh ft

### PR DESCRIPTION
**Pull request by RickRietveld**

### CODE REVIEW

File: navmesh-localized/Assets/Scripts/ClickToMove.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        agent = GetComponent<NavMeshAgent>();
+    }
+
+    void Update()
+    {
+        if (Input.GetMouseButtonDown(0) && !Input.GetKey(KeyCode.LeftShift))
+        {
+            var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+            if (Physics.Raycast(ray.origin, ray.direction, out hitInfo))
+                agent.destination = hitInfo.point;
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Scripts/ClickToMove.cs]: Hier moeten brackets omheen, zoals aangegeven wordt in de style guide

File: navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        {
+            if (GUILayout.Button("Select the Prefab asset to bake or clear the navmesh", EditorStyles.helpBox))
+            {
+                Selection.activeObject = PrefabUtility.GetCorrespondingObjectFromSource(go);
+                EditorGUIUtility.PingObject(Selection.activeObject);
+            }
+        }
+
+        if (string.IsNullOrEmpty(path))
+            return;
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs]: Hier moeten brackets, zoals aangegeven in de style guide.

File: navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+        GUILayout.BeginHorizontal();
+        GUILayout.Space(EditorGUIUtility.labelWidth);
+
+        if (GUILayout.Button("Clear"))
+            OnClear();
+
+        if (GUILayout.Button("Bake"))
+            OnBake();
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs]: idem.

File: navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    void DestroyNavMeshData(string path)
+    {
+        // Destroy and remove all existing NavMeshData sub-assets
+        var assets = AssetDatabase.LoadAllAssetsAtPath(path);
+        foreach (var o in assets)
+        {
+            var data = o as NavMeshData;
+            if (data != null)
+                DestroyImmediate(o, true);
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs:142]: idem.

File: navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        }
+    }
+
+    [DrawGizmo(GizmoType.Selected | GizmoType.Active | GizmoType.Pickable)]
+    static void RenderGizmo(NavMeshPrefabInstance instance, GizmoType gizmoType)
+    {
+        if (!EditorApplication.isPlaying)
+            instance.UpdateInstance();
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs:150]: idem.

File: navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    void OnEnable()
+    {
+        // Construct and add navmesh
+        navMesh = new NavMeshData();
+        instance = NavMesh.AddNavMeshData(navMesh);
+        if (tracked == null)
+            tracked = transform;
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs]: idem.

File: navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    void UpdateNavMesh(bool asyncUpdate = false)
+    {
+        NavMeshSourceTag.Collect(ref sources);
+        var defaultBuildSettings = NavMesh.GetSettingsByID(0);
+        var bounds = QuantizedBounds();
+
+        if (asyncUpdate)
+            operation = NavMeshBuilder.UpdateNavMeshDataAsync(navMesh, defaultBuildSettings, sources, bounds);
+        else
+            NavMeshBuilder.UpdateNavMeshData(navMesh, defaultBuildSettings, sources, bounds);
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs]: idem.

File: navmesh-localized/Assets/Scripts/NavMeshPrefabInstance.cs [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    public static List<NavMeshPrefabInstance> trackedInstances {get {return trackedInstancesList; }}
+    Vector3 position;
+    Quaternion rotation;
+
+    void OnEnable()
+    {
+        AddInstance();
+
+        if (instance.valid && followTransformToggle)
+            AddTracking();
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Scripts/NavMeshPrefabInstance.cs]: idem.

File: navmesh-localized/Assets/Scripts/NavMeshPrefabInstance.cs [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    {
+#if UNITY_EDITOR
+        if (instance.valid)
+        {
+            Debug.LogError("Instance is already added: " + this);
+            return;
+        }
+#endif
+        if (navMesh)
+            instance = NavMesh.AddNavMeshData(navMesh, transform.position, transform.rotation);
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Scripts/NavMeshPrefabInstance.cs]: idem

File: navmesh-localized/Assets/Scripts/NavMeshPrefabInstance.cs [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+                instance.UpdateInstance();
+        }
+    }
+
+#if UNITY_EDITOR
+    void OnValidate()
+    {
+        // Only when the instance is valid (OnEnable is called) - we react to changes caused by serialization
+        if (!instance.valid)
+            return;
```
Comment (SeanSnel) [inline navmesh-localized/Assets/Scripts/NavMeshPrefabInstance.cs]: idem.

**Review by SeanSnel — CHANGES_REQUESTED** [review]
Niet volledig volgens de style guide. De inhoud is echter wel goed!

**Comment by SeanSnel** [comment]
[NPC pathfinding](https://trello.com/c/f0fQaJzy/2-npc-pathfinding)

**Review by SeanSnel — DISMISSED** [review]
Ziet er goed uit!

File: navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    public void OnEnable()
+    {
+        followTransformProp = serializedObject.FindProperty("m_FollowTransform");
+        navMeshDataProp = serializedObject.FindProperty("m_NavMesh");
+    }
+
+    public override void OnInspectorGUI()
+    {
+        var instance = (NavMeshPrefabInstance)target;
+        var go = instance.gameObject;
```
Comment (marthijn13) [inline navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs:22]: You are getting an instanced gameobject, why use var?

File: navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+        navmesh.name = "Navmesh";
+        return navmesh;
+    }
+
+    void OnClear()
+    {
+        foreach (var tgt in targets)
+        {
+            var instance = (NavMeshPrefabInstance)tgt;
+            var go = instance.gameObject;
```
Comment (marthijn13) [inline navmesh-localized/Assets/Editor/NavMeshPrefabInstanceEditor.cs:93]: You are getting an instanced gameobject, why use var?

File: navmesh-localized/Assets/Scripts/ClickToMove.cs [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    void Start()
+    {
+        agent = GetComponent<NavMeshAgent>();
+    }
+
+    void Update()
+    {
+        if (Input.GetMouseButtonDown(0) && !Input.GetKey(KeyCode.LeftShift))
+        {
+            var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
```
Comment (marthijn13) [inline navmesh-localized/Assets/Scripts/ClickToMove.cs:20]: var

File: navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+public class LocalNavMeshBuilder : MonoBehaviour
+{
+    // The center of the build
+    [SerializeField]
+    private Transform tracked;
+    [SerializeField]
+    private bool updateNavMesh;
+    // The size of the build bounds
+    [SerializeField]
+    private Vector3 size = new Vector3(80.0f, 20.0f, 80.0f);
```
Comment (marthijn13) [inline navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs:18]: Magic numbers, can these be placed in variables and serialized so they are easily changable?

**Review by marthijn13 — CHANGES_REQUESTED** [review]

**Comment by marthijn13** [comment]
A lot of vars that dont need to be there

File: navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs (hunk 14 above)
Comment (RickRietveld) [reply navmesh-localized/Assets/Scripts/LocalNavMeshBuilder.cs:18]: These numbers are editable in the inspector -> (size is serialized)

**Review by SeanSnel — APPROVED** [review]

**Review by marthijn13 — APPROVED** [review]

**Pull request merged**